Add stack canaries to both the payload and the loader

MikeIsAStar opened the pull request.

stblr commented on common/StackCanary.h:
```diff
@@ -0,0 +1,6 @@
+
+#include "common/Types.h"
+
+asm void StackCanary_Init(void);
+asm u32 StackCanary_Canary(void);
```
Can the `asm` part be moved to the c file?

MikeIsAStar replied:
Resolved in commit [7716b67](https://github.com/doubledashdeluxe/ddd/pull/66/commits/7716b677b06aafacbcc399c725836f8cdc6ee1dc).

stblr reviewed (commented):
The commit message mentions the payload, but I'm pretty sure they are also added to the loader.

MikeIsAStar commented:
> The commit message mentions the payload, but I'm pretty sure they are also added to the loader.

You are absolutely correct ! I have updated the commit message to reflect this.
